main.cpp

```diff
@@ -120,9 +205,20 @@ struct CommonTreasureChest
     bool isRare = false;
     bool isOpened = false;
 
-    void openChest(bool isOpened);
+    bool openChest(bool openState);
 };
 
+bool CommonTreasureChest::openChest (bool openState)
+{
+    if (openState == false)
+    {
+        //triger function to open the Box
+        return true;
+    }
+
```

Comment:
You must return something. 
You could simplify this whole function by writing: 
`return (openState == false);` and eliminate the `if()` block entirely. 
- [ ] simplify